HW2/P5/driver.py

```diff
@@ -80,3 +110,19 @@ def randcolor():
             # SUBPROBLEM 3: sort objects by location.  Be sure to update the
             # grid if objects' indices change!  Also be sure to sort the
             # velocities with their object positions!
+
+            # Assign weights to each ball
+            loc = [((int)(ball[0]/grid_spacing), (int)(ball[1]/grid_spacing)) for ball in positions]
+            weight = [weight_grid[l[0], l[1]] if (l[0]<weight_grid.shape[0] and l[1]<weight_grid.shape[1]) else 0 for l in loc]
+
+            idx = np.argsort(weight)
+
+            # reset positions and velocities
+            positions = positions[idx]
+            velocities = velocities[idx]
+            # reset grid
+            grid[:,:] = -1
+            for i in range(num_balls):
+                if positions[i,0]<1 and positions[i,1]<1:
+                    grid[(int)(positions[i,0]/grid_spacing), (int)(positions[i,1]/grid_spacing)] = i
+
```

Comment:
Is the correct procedure to not assign a grid space to balls out of bounds or is it to put them in the closest valid grid space? (Or is either fine?)